public/app/core/utils/CorsSharedWorker.ts

```diff
@@ -0,0 +1,17 @@
+// works with webpack plugin: scripts/webpack/plugins/CorsWorkerPlugin.js
+export class CorsSharedWorker extends SharedWorker {
+  constructor(url: URL, options?: WorkerOptions) {
+    // by default, worker inherits HTML document's location and pathname which leads to wrong public path value
+    // the CorsWorkerPlugin will override it with the value based on the initial worker chunk, ie.
+    //    initial worker chunk: http://host.com/cdn/scripts/worker-123.js
+    //    resulting public path: http://host.com/cdn/scripts
+
+    const scriptUrl = url.toString();
+    const urlParts = scriptUrl.split('/');
+    urlParts.pop();
+    const scriptsBasePathUrl = `${urlParts.join('/')}/`;
+
+    const source = `__webpack_worker_public_path__ = '${scriptsBasePathUrl}'; importScripts('${scriptUrl}');`;
+    super(`data:application/javascript;base64,${btoa(source)}`, options);
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/Security/Same-origin_policy#inherited_origins this won't work because origin of the loaded worker will be "null"